### PR TITLE
review exported names in collections stubs

### LINKED ISSUE #709: Collections stubs should use 'as' import
`stdlib/[23]/collections.pyi` should use `from typing import ..., Iterable as Iterable, ...` so that the ABCs can be imported using `from collections import *`.

The Python 3 stub already does this for some ABCs but there are missing ones.

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Fixes #709

I looked at dir(collections) in 2.7 and 3.6 and made sure the list of names matched the stubs.

### CODE REVIEW

File: stdlib/2/collections.pyi [hunk 1]
```diff
@@ -2,16 +2,28 @@
...
+# These are not exported.
+from typing import Any, Dict, Generic, TypeVar, Tuple, overload, Type, Optional, List, Union, Reversible
 
+# These are exported.
 from typing import (
-    Any, Container, Dict, Generic, TypeVar, Iterable, Tuple, Callable, Mapping, overload,
-    Iterator, Type, Sized, Optional, List, Set, Sequence, Union, Reversible,
-    MutableMapping, MutableSet, MutableSequence,
+    Callable as Callable,
+    Container as Container,
```
Comment (ambv) [inline stdlib/2/collections.pyi]: `Collection` is missing, add it too, please!

File: stdlib/3/collections/__init__.pyi [hunk 2]
```diff
@@ -2,29 +2,25 @@
...
-    Callable, Any, Type, Optional, Union
+    Any, Type, Optional, Union
 )
 # These are exported.
-# TODO reexport more.
+from . import abc
+
 from typing import (
+    Callable as Callable,
     Container as Container,
```
Comment (ambv) [inline stdlib/3/collections/__init__.pyi]: Same here, Collection is missing.

File: stdlib/2/collections.pyi (hunk 1 above)
Comment (JelleZijlstra) [inline stdlib/2/collections.pyi]: It doesn't exist, it was added in 3.6.

File: stdlib/3/collections/__init__.pyi (hunk 2 above)
Comment (JelleZijlstra) [inline stdlib/3/collections/__init__.pyi]: It's there below under the `if sys.version_info >= (3, 6)`.

**Pull request merged**